cypress/integration/component-interaction.spec.js

```diff
@@ -0,0 +1,144 @@
+const API_URL = Cypress.env( 'API_URL' )
+	.replace( /(^')|('$)/g, '' ); // for some reason Cypress keeps quotes around env vars coming from docker through `.env`
+
+function wikibaseApiRequest( query ) {
+	return {
+		url: API_URL,
+		query,
+	};
+}
+
+describe( 'Component interaction test', () => {
+    it( 'Tests whether components behave as they should when values change', () => {
+        cy.visit( '/' );
+
+        /**
+         * The intercepts basically mock an API request, as we type into the lookup component
+         * with different search strings. see the .json files in the /fixtures directory
+         */
+        cy.intercept(
+           wikibaseApiRequest( { action: 'wbsearchentities', search: 'has pet' } ),
+            { fixture: 'wbsearchentities-has-pet.json' },
+        ).as( 'hasPetRequest' );
+        cy.intercept(
+            wikibaseApiRequest( { action: 'wbsearchentities', search: 'house cat' } ),
+            { fixture: 'wbsearchentities-house-cat.json' },
+        ).as( 'houseCatRequest' );
+
+        cy.intercept(
+			wikibaseApiRequest( { action: 'wbsearchentities', search: 'instance of' } ),
+			{ fixture: 'wbsearchentities-instance-of.json' },
+		).as( 'hasInstanceRequest' );
+
+        cy.intercept(
+			wikibaseApiRequest( { action: 'wbsearchentities', search: 'IMDb ID' } ),
+			{ fixture: 'wbsearchentities-external-id.json' },
+		).as( 'hasImdbRequest' );
+
+        cy.intercept(
+            wikibaseApiRequest( { action: 'wbsearchentities', search: 'installed capacity' } ),
+            { fixture: 'wbsearchentities-limited-support.json' },
+        ).as( 'hasLimitedSupportedRequest' );
+
+        cy.intercept(
+            wikibaseApiRequest( { action: 'wbsearchentities' } ),
+            { fixture: 'wbsearchentities-empty.json' },
+        );
+
+        cy.get( '.query-condition__property-lookup .wikit-Input' )
+            .type( 'has pet' )
+            .wait( '@hasPetRequest' );
+        cy.get( '.query-condition__property-lookup .wikit-OptionsMenu__item' ).click();
+
+        cy.get( '.query-condition__value-input .wikit-Input' )
+            .type( 'house cat' )
+            .wait( '@houseCatRequest' );
+        cy.get( '.query-condition__value-input .wikit-OptionsMenu__item' )
+            .first()
+            .click();
+
+        // clear out values from specified input components
+        cy.get( '.query-condition__property-lookup .wikit-Input' ).clear();
```

Comment:
good feedbacks :)

- I added a cypress lint plugin and added some rules for linting.
- I agree with the other two tests. they are kinda redundant at these point. and the two new test are pretty comprehensive and cover all the scenarios the old tests were doing. I deleted them.
- I'm trying to cover more 'realistic' user interaction scenarios. all the 4 conditions ((string, item, external-id, limited support) ) have been covered in different scenarios.

Reply:
Not sure I follow entirely, especially because I didn't see the string datatype, but this is already much better than what we had before.
Maybe somebody else can add more to the browser test so that we spread the knowledge of writing Cypress browser tests around.

Reply:
I agree. we should write more browser tests (for upcoming components) to spread the knowledge. 
I was actually talking about the `@hasPetRequest`. its a string search.... AFAIK,

maybe we should make browser testing a part of development (like unit testing). so people can add more tests, whenever a UI task is involved